24-common-raven/technical_specification.md

```diff
@@ -14,17 +14,26 @@ This epic covers the following user journeys:
 
 Produce a script to test the download path in the testbed and adapt the DCS and IFRS to allow for a fully functionioning download path.
 
+### Re-Encryption
+
+Add a new S3 - Bucket called "Staging", there the IRS has write and the IFRS has read permissions. The IRS is now re-encrypting all files during checksum validation and the file is staged to the staging bucket.
+
 ## Additional Implementation Details:
 
 ### IFRS
 - The IFRS needs to consume the *NonStagedFileRequested* event, stage file to outbox and emit the *FileStagedForDownload* event afterwards
-
-### IFRS/Hexkit
-- Implement multipart copy with offset to store file object without envelope in permanent storage
+- Copy the encrypted, headless file from the staging bucket to the permanent storage after "interrogation_successfull"
 
 ### DCS
 - Adapt to request envelope from EKSS and serve the assembled file (envelope + encrypted file content) using custom download ranges
 
+### EKSS
+- When the IRS sends the first part, the EKSS now creates a new secret (using os.urandom32), and stores the new, instead of the old secret in vault. Both secrets are then sent back.
```

Comment:
```suggestion
- When the IRS sends the first part, the EKSS now creates a new secret (using os.urandom(32) as per crypt4gh), and stores the new, instead of the old secret in vault. Both secrets are then sent back along with the ID generated for the new secret
```